docs/gadgets/prometheus.md

```diff
@@ -0,0 +1,364 @@
+namespace/monitoring created
+serviceaccount/prometheus created
+clusterrole.rbac.authorization.k8s.io/discoverer created
+clusterrolebinding.rbac.authorization.k8s.io/prometheus-discoverer created
+configmap/prometheus-server-conf created
+deployment.apps/prometheus created
+```
+
+Create a port-forward session to Prometheus:
+
+```bash
+$ kubectl port-forward --namespace monitoring deployment/prometheus 9090:9090 &
+```
+
+Let's create a metric that reports processes executed:
+
+```yaml
+# myconfig.yaml
+metrics:
+  - name: executed_processes
+    type: counter
+    category: trace
+    gadget: exec
+    labels:
+      - namespace
+      - pod
+      - container
+```
+
+Start the gadget
+
+```bash
+$ kubectl gadget prometheus --config @myconfig.yaml
+INFO[0000] minikube-m02         | Running. Metrics will be updated while this gadget is running
+INFO[0000] minikube             | Running. Metrics will be updated while this gadget is running
+```
+
+Now, the `executed_processes_total` counter is available in Prometheus:
+
+![Inspektor Gadget Counter Metric](../images/prometheus_counter_1.png)
+
+You can see that the counters are already going up for some containers.
+
+Let's create a pod to execute from more processes:
+
+```bash
+$ kubectl run mypod1 -it --image busybox --restart Never -- sh -c 'for i in $(seq 0 1 1000); do cat /dev/null ; ping -c 1 localhost > /dev/null; done'
+```
+
+If we check the counter again, we can see that it shows that our pod has executed a lot of processes:
+
+![Inspektor Gadget Counter Metric](../images/prometheus_counter_2.png)
+
+Now, update the configuration file to only take into considerations executions of the `cat` binary:
+
+```yaml
+metrics:
+  - name: executed_processes
+    type: counter
+    category: trace
+    gadget: exec
+    labels:
+      - namespace
+      - pod
+      - container
+    selector:
+     - "comm:cat"
+```
+
+Restart the gadget
+
+```bash
+$ kubectl gadget prometheus --config @myconfig.yaml --metrics-path /metrics
+INFO[0000] minikube-m02         | Running. Metrics will be updated while this gadget is running
+INFO[0000] minikube             | Running. Metrics will be updated while this gadget is running
+```
+
+Create a new pod that executes processes:
+
+```bash
+$ kubectl run mypod2 -it --image busybox --restart Never -- sh -c 'for i in $(seq 0 1 1000); do cat /dev/null ; ping -c 1 localhost > /dev/null; done'
+
+```
+
+The counter only takes into consideration the cat commands now:
+
+![Inspektor Gadget Counter Metric](../images/prometheus_counter_3.png)
+
+#### With `ig`
+
+It's also possible to use the prometheus gadget without Kubernetes. In this case, we have to
+configure Prometheus to point to the endpoint exposed by ig, it's `localhost:2223` by default:
+
+```yaml
+# prometheus.yaml
+scrape_configs:
+- job_name: ig
+  scrape_interval: 1s
+  static_configs:
+  - targets:
+    - localhost:2223
+```
+
+Start prometheus with such configuration
```

Comment:
```suggestion
Start prometheus with above configuration (please refer to [docker installation](https://prometheus.io/docs/prometheus/latest/installation/#using-docker) in case you want to run prometheus in a container)
```